pkg/serviceaccount/legacy.go

```diff
@@ -42,3 +48,89 @@ type legacyPrivateClaims struct {
 	SecretName         string `json:"kubernetes.io/serviceaccount/secret.name"`
 	Namespace          string `json:"kubernetes.io/serviceaccount/namespace"`
 }
+
+func NewLegacyValidator(lookup bool, getter ServiceAccountTokenGetter) Validator {
+	return &legacyValidator{
+		lookup: lookup,
+		getter: getter,
+	}
+}
+
+type legacyValidator struct {
+	lookup bool
+	getter ServiceAccountTokenGetter
+}
+
+var _ = Validator(&legacyValidator{})
+
+func (v *legacyValidator) Validate(tokenData string, public *jwt.Claims, privateObj interface{}) (user.Info, error) {
```

Comment:
straight move?

Reply:
Everything except for the return is the same. The function new returns info.User, so there are a lot more nils, and one user.Info returned at the end.